java/GENEActivParser.java

```diff
@@ -0,0 +1,229 @@
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+
+public class GENEActivParser {
+
+    final private static int EXIT_SUCCESS = 0;
+    final private static int EXIT_FAILURE = 1;
+
+    // Specification of items to be written
+    private static final Map<String, String> ITEM_NAMES_AND_TYPES;
+    static{
+        Map<String, String> itemNamesAndTypes = new LinkedHashMap<String, String>();
+        itemNamesAndTypes.put("time", "Long");
+        itemNamesAndTypes.put("x", "Double");
+        itemNamesAndTypes.put("y", "Double");
+        itemNamesAndTypes.put("z", "Double");
+        itemNamesAndTypes.put("T", "Double");
+        ITEM_NAMES_AND_TYPES = Collections.unmodifiableMap(itemNamesAndTypes);
+    }
+
+
+    public static int parse(
+        String accFile,
+        String outFile,
+        boolean verbose) {
+
+        int fileHeaderSize = 59;
+        int linesToAxesCalibration = 47;
+        int pageHeaderSize = 9;
+        int[] errCounter = new int[] { 0 };
+
+        NpyWriter writer = new NpyWriter(outFile, ITEM_NAMES_AND_TYPES);
+
+        try {
+            BufferedReader rawAccReader = new BufferedReader(new FileReader(accFile));
```

Comment:
It would be good to continue to have the current functionality of directly reading .gz files